x-pack/plugins/observability_solution/dataset_quality/server/utils/reduce_async_chunks.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Observable, OperatorFunction, from, lastValueFrom, mergeMap, reduce } from 'rxjs';
+import deepmerge from 'deepmerge';
+
+type CallbackFn<TResult> = (chunk: string[], id: number) => Promise<TResult>;
+
+const MAX_HTTP_LINE_LENGTH = 4096;
+// Apply an 80% threshold to the http line max length to guarantee enough space for url and potentially other parameters.
+// This value might need to vary as it's an estimate of how much we can reserve for the chunked list length.
+const MAX_CHUNK_LENGTH = MAX_HTTP_LINE_LENGTH * 0.75; // 4096 *0.75 === 3072 characters, as 1 chars = 1 byte
+
+export const reduceAsyncChunks = <TResult>(list: string[], chunkExecutor: CallbackFn<TResult>) => {
+  const result$ = from(list).pipe(
+    bufferUntil(isLessThanMaxChunkLength),
+    mergeMap((chunk, id) => from(chunkExecutor(chunk, id))),
+    reduce((result, chunkResult) => deepmerge(result, chunkResult))
+  );
+
+  return lastValueFrom(result$);
+};
+
+/**
+ * Support functions for reduceAsyncChunks
+ */
+const bufferUntil = <TItem>(
```

Comment:
> [!NOTE]
> This function supports the steps piped in the resolution performed by `reduceAsyncChunks`.
It returns a `rxjs` operator that, given a predicate callback as an argument that takes the current chunk and the current Item from the list, uses this callback predicate to compute whether the current item fits in the current chunk, or whether we need to consider the current chunk filled and start a new one.